### PR TITLE
envd: Rename BOOTSTRAP_BUILTIN_INTROSPECTION_CLUSTER_REPLICA_SIZE

### PR DESCRIPTION
**Pull request by def-**
to BOOTSTRAP_BUILTIN_CATALOG_SERVER_CLUSTER_REPLICA_SIZE

Follow-up to https://github.com/MaterializeInc/materialize/issues/26731

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/A

### CODE REVIEW

**Comment by teskje** [comment]
Sorry, forgot about that TODO 🙈 

This change needs to be synchronized with the cloud repo. The environment controller uses this flag when spawning envd containers and needs to switch to the new one for the version of Mz that will have the rename.

**Comment by def-** [comment]
@teskje Done: https://github.com/MaterializeInc/cloud/pull/9592 I'm out next week, do you want to take this over?

**Comment by teskje** [comment]
For sure!

**Review by teskje — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
@teskje, if you are confident with your changes and the one in the cloud repo, feel free to take over this PR and merge it. Dennis is out this week.

**Comment by teskje** [comment]
Yep! Just waiting for approval on the cloud PR so I can merge both at the same time.

**Pull request merged**
